src/arrayMethodJoin.js

```diff
@@ -4,9 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      resultString += i === 0
```

Comment:
Strange condition. Why didn't you use it?
`
if (i < this.length - 1) {
  nextElement(this[i]) + separator;
} else {
  nextElement(this[i])
}
`